src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
 export const App: React.FC = () => {
+  const [items, setItems] = useState<string[]>(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [initialItems] = useState<string[]>(goodsFromServer);
+
+  function sortByAlphabet() {
+    if (sortType === SortType.Alphabet && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((a, b) => a.localeCompare(b));
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Alphabet);
+  }
```

Comment:
The function name 'sortByAlphabet' does not start with a verb that indicates its action. Consider renaming it to clearly reflect the action it performs, such as 'handleAlphabetSort'. This follows the good example provided in the checklist.